frontend/public/module/k8s/k8s-models.ts

```diff
@@ -5,20 +5,37 @@ import { K8sResourceKindReference, K8sKind } from './index';
 import * as staticModels from '../../models';
 import { referenceForModel, kindForReference } from './k8s';
 import store from '../../redux';
+import * as plugins from '../../plugins';
+
+const getAllStaticModels = () => {
+  let models = [ ..._.values(staticModels) ];
+
+  // auto-detect additional models from known Console extensions
+  plugins.registry.getFeatureFlags().filter(plugins.isModelFeatureFlag).forEach(ext => {
+    models = [ ...models, ext.properties.model ];
+  });
+  plugins.registry.getResourcePages().forEach(ext => {
+    models = [ ...models, ext.properties.model ];
+  });
+
+  return _.uniqWith(models, (a, b) => referenceForModel(a) === referenceForModel(b));
```

Comment:
Do we expect multiple plugins to define the same model? I feel like we should at least print a warning, particularly if the models aren't exactly the same.

Reply:
I believe this can be

```suggestion
  return _.uniqWith(models, referenceForModel);
```

Reply:
Disregard my second comment, misread the docs :)

Reply:
That's a good point.

A plugin should not re-define existing static models, this includes both core Console models (`public/models/index.ts`) and models contributed by other plugins (e.g. `BaremetalHostModel` from Metal3 plugin).

> I feel like we should at least print a warning, particularly if the models aren't exactly the same.

Agreed, I'll adapt the code accordingly.

Reply:
PR updated, conflicting models will be discarded along with a console log warning.